src/main/java/com/vaadin/guice/server/VaadinModule.java

```diff
@@ -39,6 +40,7 @@ public VaadinModule(SessionProvider sessionProvider, Set<Class<? extends View>>
     @Override
     protected void configure() {
         bindScope(UIScope.class, uiScoper);
+        bindScope(GuiceUI.class, uiScoper);
```

Comment:
I think that it makes sense to have the UI added in the UI-scope. A possible use case would be to have a specific UI implementation injected somewhere (or a provider for that specific UI). I would expect that UI instance to be the 'singleton' UI within the current UI scope, not a brand new instance.